dll/win32/kernel32/client/fiber.c

```diff
@@ -376,44 +427,24 @@ WINAPI
 FlsSetValue(DWORD dwFlsIndex,
             PVOID lpFlsData)
 {
+    PPEB Peb = NtCurrentPeb();
     PVOID *ppFlsSlots;
-    TEB *pTeb = NtCurrentTeb();
-
-    if(dwFlsIndex >= 128) goto l_InvalidParam;
-
-    ppFlsSlots = pTeb->FlsData;
 
-    if (ppFlsSlots == NULL)
+    if (!dwFlsIndex || dwFlsIndex >= FLS_MAXIMUM_AVAILABLE)
     {
-        PEB *pPeb = pTeb->ProcessEnvironmentBlock;
-
-        ppFlsSlots = RtlAllocateHeap(pPeb->ProcessHeap,
-                                     HEAP_ZERO_MEMORY,
-                                     (128 + 2) * sizeof(PVOID));
-        if(ppFlsSlots == NULL) goto l_OutOfMemory;
-
-        pTeb->FlsData = ppFlsSlots;
-
-        RtlAcquirePebLock();
-
-        /* TODO: initialization */
-
-        RtlReleasePebLock();
+        SetLastError(ERROR_INVALID_PARAMETER);
+        return FALSE;
     }
-
+    if (!NtCurrentTeb()->FlsData &&
+        !(NtCurrentTeb()->FlsData = RtlAllocateHeap(RtlGetProcessHeap(), HEAP_ZERO_MEMORY,
```

Comment:
Cannot we use ppFlsSlots here instead, and then if the allocation succeeds, assign ppFlsSlots to NtCurrentTeb()->FlsData() ?